src/App.tsx

```diff
@@ -1,73 +1,133 @@
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import classNames from 'classnames';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Props = {
+  debounceDelay?: number;
+  onSelected?: (person: Person | null) => void;
+};
+
+export const App: React.FC<Props> = ({ debounceDelay = 300, onSelected }) => {
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+  const [debouncedQuery, setDebouncedQuery] = useState(query);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedQuery(query);
+    }, debounceDelay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [query, debounceDelay]);
+
+  const filteredPeople = useMemo(() => {
+    if (debouncedQuery === '') {
+      return peopleFromServer;
+    }
+
+    const filteredArr = peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(debouncedQuery.toLowerCase()),
+    );
+
+    setError(filteredArr.length === 0);
+
+    return filteredArr;
+  }, [debouncedQuery]);
+
+  const searchByName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const searchValue = event.target.value;
+
+    setQuery(prevQuery => {
+      if (prevQuery !== searchValue) {
+        setSelectedPerson(null);
+
+        if (onSelected) {
+          onSelected(null);
+        }
+      }
+
+      return searchValue;
+    });
+  };
+
+  const handlePersonClick = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setIsDropdownActive(false);
+    if (onSelected) {
+      onSelected(person);
+    }
+  };
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {selectedPerson
+            ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+            : 'No selected person'}
         </h1>
 
-        <div className="dropdown is-active">
+        <div
+          className={classNames('dropdown', { 'is-active': isDropdownActive })}
+        >
           <div className="dropdown-trigger">
             <input
               type="text"
               placeholder="Enter a part of the name"
               className="input"
               data-cy="search-input"
+              onChange={searchByName}
+              value={query}
+              onFocus={() => {
+                setIsDropdownActive(true);
+                if (query.trim() === '') {
+                  setError(false);
+                }
+              }}
+              onBlur={() => {
+                setTimeout(() => setIsDropdownActive(false), debounceDelay);
+              }}
```

Comment:
The use of `setTimeout` in the `onBlur` event handler to set `isDropdownActive` to `false` might cause a delay that could lead to unexpected behavior if the user quickly interacts with the input field again. Consider using a more immediate approach to handle the dropdown state change.